packages/plugin-ext/src/plugin/plugin-context.ts

```diff
@@ -972,19 +972,33 @@ export function createAPIFactory(
     };
 }
 
-class Plugin<T> implements theia.Plugin<T> {
+enum ExtensionKind {
+    UI = 1,
+    Workspace = 2
+}
+
+export class PluginInstance<T> implements theia.Plugin<T> {
```

Comment:
Is the renaming of this class necessary? Changing the name of an export is a breaking change, so if it isn't strictly necessary, it's better to avoid it.

Reply:
Good point! I changed it back to the original class name though the original class was not exported so I guess it never was officially API. I now export the "base" Plugin class as well as the extended PluginExt class so that adopters could even base their own implementation on top of that if they need to.